### PR TITLE
Remove TODO in util/testing for IsTrueWithin()

### LINKED ISSUE #4633: circleci: failed tests (13386): TestBaseQueuePurgatory
The following test appears to have failed:

[#13386](https://circleci.com/gh/cockroachdb/cockroach/13386):

```
--- PASS: TestBaseQueueAddRemove (0.00s)
=== RUN   TestAcceptsUnsplitRanges
I160225 05:01:11.379339 storage/queue.go:265  [test] range=2 [/Table/11-/Max): split needed; not adding
--- PASS: TestAcceptsUnsplitRanges (0.01s)
=== RUN   TestBaseQueuePurgatory
--- FAIL: TestBaseQueuePurgatory (0.29s)
	queue_test.go:499: expected purgatory size of 10; got 9
	<autogenerated>:32: storage/queue_test.go:515: condition failed to evaluate within 250ms: expected 20 processed replicas; got 19
=== RUN   TestGetTruncatableIndexes
I160225 05:01:11.678848 raft/raft.go:459  [group 1] 1 became follower at term 5
I160225 05:01:11.679003 raft/raft.go:235  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I160225 05:01:11.679071 raft/raft.go:539  [group 1] 1 is starting a new election at term 5
I160225 05:01:11.679143 raft/raft.go:472  [group 1] 1 became candidate at term 6
I160225 05:01:11.679206 raft/raft.go:521  [group 1] 1 received vote from 1 at term 6
I160225 05:01:11.679288 raft/raft.go:500  [group 1] 1 became leader at term 6
I160225 05:01:11.681332 storage/replica.go:1630  [node=1,store=1,range=1] gossiping cluster id "00000000-0000-0000-0000-000000000000" from store 1, range 1
--
W160225 05:01:45.132363 sql/pgwire/v3.go:154  unrecognized configuration parameter "datestyle"
I160225 05:01:45.158645 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestLogRebalances (0.67s)
=== RUN   Example_rebalancing
--- PASS: Example_rebalancing (0.97s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	35.408s
=== RUN   TestBatchBasics
I160225 05:01:24.547700 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchGet
I160225 05:01:24.556080 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.02s)
=== RUN   TestBatchMerge
I160225 05:01:24.577869 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.01s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

### PR DESCRIPTION
**Pull request by spencerkimball**
Replace all instances of IsTrueWithin with SucceedsWithin.
Change SucceedsWithin to always use a 15s duration.

Fixes #4633

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4724)

### CODE REVIEW

File: acceptance/gossip_peerings_test.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ type checkGossipFunc func(map[string]interface{}) error
 // retrying for up to the given duration.
 func checkGossip(t *testing.T, c cluster.Cluster, d time.Duration,
 	f checkGossipFunc) {
-	util.SucceedsWithin(t, d, func() error {
+	util.SucceedsWithin(t, func() error {
```
Comment (bdarnell) [inline acceptance/gossip_peerings_test.go]: Argument `d` is no longer used.

File: acceptance/replication_test.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ func checkRangeReplication(t *testing.T, c cluster.Cluster, d time.Duration) {
 
 	log.Infof("waiting for first range to have %d replicas", wantedReplicas)
 
-	util.SucceedsWithin(t, d, func() error {
+	util.SucceedsWithin(t, func() error {
```
Comment (bdarnell) [inline acceptance/replication_test.go]: Ditto

**Comment by bdarnell** [comment]
The name `SucceedsWithin` doesn't make sense now that it's not `SucceedsWithin(duration)`. `SucceedsSoon`? `AssertSoon`?

LGTM

**Comment by tschottdorf** [comment]
LGTM and ditto what @bdarnell said.

File: acceptance/gossip_peerings_test.go (hunk 1 above)
Comment (spencerkimball) [inline acceptance/gossip_peerings_test.go]: Done.

File: acceptance/replication_test.go (hunk 2 above)
Comment (spencerkimball) [inline acceptance/replication_test.go]: Done.

**Comment by spencerkimball** [comment]
Renamed to `SucceedsSoon`.

**Pull request merged**